### ISSUE TITLE
(Rootless) Pod that is connected to network does not run after reboot

### ISSUE THREAD

**Issue by phongngthanh** [containers/ansible-podman-collections#248]
**Is this a BUG REPORT or FEATURE REQUEST? (leave only one on its own line)**

/kind bug

**Description**

(Rootless) Pod that is connected to a named network cannot run without network recreation after reboot

**Steps to reproduce the issue:**

1. Run the following playbook:
```yaml
- name: Create network
  containers.podman.podman_network:
    name: test_network

- name: Create pod
  containers.podman.podman_pod:
    name: test_pod
    network: test_network
    ports:
      - '8000:80'

- name: Run test
  containers.podman.podman_container:
    name: test-container
    image: containous/whoami:latest
    pod: test_pod
    restart_policy: "unless-stopped"
    state: present
```

2. Container runs normally. Reboot the computer.

3. Rerun the playbook.

**Describe the results you received:**
1. Container recreation failed
```bash
TASK [bug : Create network] ****************************************************
ok: [localhost]

TASK [bug : Create pod] ********************************************************
ok: [localhost]

TASK [bug : Run test] **********************************************************
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Can't run container test-container", "stderr": "time=\"2021-04-24T09:32:04+07:00\" level=error msg=\"error starting some container dependencies\"\ntime=\"2021-04-24T09:32:04+07:00\" level=error msg=\"\\\"command rootless-cni-infra [alloc 2eb907fa3e5ec94acd05ce9d449391d125d60363eaf34b79df75f04ac22fb131 test_network test_pod   ] in container 995ce4d5108faed62de42510df329d649674dc7af04fc9acc1d9c2a3dd18422f failed with status 1, stdout=\\\\\\\"\\\\\\\", stderr=\\\\\\\"failed to open netns \\\\\\\\\\\\\\\"/proc/19057/ns/net\\\\\\\\\\\\\\\": failed to Statfs \\\\\\\\\\\\\\\"/proc/19057/ns/net\\\\\\\\\\\\\\\": no such file or directory\\\\\\\\n\\\\\\\"\\\"\"\nError: error starting some containers: internal libpod error\n", "stderr_lines": ["time=\"2021-04-24T09:32:04+07:00\" level=error msg=\"error starting some container dependencies\"", "time=\"2021-04-24T09:32:04+07:00\" level=error msg=\"\\\"command rootless-cni-infra [alloc 2eb907fa3e5ec94acd05ce9d449391d125d60363eaf34b79df75f04ac22fb131 test_network test_pod   ] in container 995ce4d5108faed62de42510df329d649674dc7af04fc9acc1d9c2a3dd18422f failed with status 1, stdout=\\\\\\\"\\\\\\\", stderr=\\\\\\\"failed to open netns \\\\\\\\\\\\\\\"/proc/19057/ns/net\\\\\\\\\\\\\\\": failed to Statfs \\\\\\\\\\\\\\\"/proc/19057/ns/net\\\\\\\\\\\\\\\": no such file or directory\\\\\\\\n\\\\\\\"\\\"\"", "Error: error starting some containers: internal libpod error"], "stdout": "", "stdout_lines": []}

PLAY RECAP *********************************************************************
localhost                  : ok=3    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0   
```

**Describe the results you expected:**
1. Network and pods status are ok, container creation is done normally (status: changed)

**Additional information you deem important (e.g. issue happens only occasionally):**
1. The issue doesn't happen when container is connected directly to the named network, or when the pod isn't connected to a named network, or if the container, network, and pod were created by root user (become: true)
2. Recreating pod return the same error.
3. If network is recreated, the container runs normally.

**Version of the `containers.podman` collection:**
**Or version from `ansible-galaxy` if installed from galaxy: `ansible-galaxy collection list | grep containers.podman`**

```
(paste your output here)
```

**Output of `ansible --version`:**

```
ansible 2.9.18
  config file = /etc/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  executable location = /usr/bin/ansible
  python version = 3.9.2 (default, Feb 20 2021, 00:00:00) [GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]
```

**Output of `podman version`:**

```
podman version 3.1.2
```

**Output of `podman info --debug`:**

```
host:
  arch: amd64
  buildahVersion: 1.20.1
  cgroupManager: systemd
  cgroupVersion: v2
  conmon:
    package: conmon-2.0.27-2.fc33.x86_64
    path: /usr/bin/conmon
    version: 'conmon version 2.0.27, commit: '
  cpus: 8
  distribution:
    distribution: fedora
    version: "33"
  eventLogger: journald
  hostname: server
  idMappings:
    gidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
    uidmap:
    - container_id: 0
      host_id: 1000
      size: 1
    - container_id: 1
      host_id: 100000
      size: 65536
  kernel: 5.11.15-200.fc33.x86_64
  linkmode: dynamic
  memFree: 30000074752
  memTotal: 33339879424
  ociRuntime:
    name: crun
    package: crun-0.19-1.fc33.x86_64
    path: /usr/bin/crun
    version: |-
      crun version 0.19
      commit: e67a75672412975916dac6b87f8346f870e4b99a
      spec: 1.0.0
      +SYSTEMD +SELINUX +APPARMOR +CAP +SECCOMP +EBPF +CRIU +YAJL
  os: linux
  remoteSocket:
    exists: true
    path: /run/user/1000/podman/podman.sock
  security:
    apparmorEnabled: false
    capabilities: CAP_CHOWN,CAP_DAC_OVERRIDE,CAP_FOWNER,CAP_FSETID,CAP_KILL,CAP_NET_BIND_SERVICE,CAP_SETFCAP,CAP_SETGID,CAP_SETPCAP,CAP_SETUID,CAP_SYS_CHROOT
    rootless: true
    seccompEnabled: true
    selinuxEnabled: true
  slirp4netns:
    executable: /usr/bin/slirp4netns
    package: slirp4netns-1.1.9-1.fc33.x86_64
    version: |-
      slirp4netns version 1.1.9
      commit: 4e37ea557562e0d7a64dc636eff156f64927335e
      libslirp: 4.3.1
      SLIRP_CONFIG_VERSION_MAX: 3
      libseccomp: 2.5.0
  swapFree: 4294963200
  swapTotal: 4294963200
  uptime: 3m 44.63s
registries:
  search:
  - registry.fedoraproject.org
  - registry.access.redhat.com
  - registry.centos.org
  - docker.io
store:
  configFile: /home/user/.config/containers/storage.conf
  containerStore:
    number: 31
    paused: 0
    running: 0
    stopped: 31
  graphDriverName: overlay
  graphOptions:
    overlay.mount_program:
      Executable: /usr/bin/fuse-overlayfs
      Package: fuse-overlayfs-1.5.0-1.fc33.x86_64
      Version: |-
        fusermount3 version: 3.9.3
        fuse-overlayfs: version 1.5
        FUSE library version 3.9.3
        using FUSE kernel interface version 7.31
  graphRoot: /home/user/.local/share/containers/storage
  graphStatus:
    Backing Filesystem: btrfs
    Native Overlay Diff: "false"
    Supports d_type: "true"
    Using metacopy: "false"
  imageStore:
    number: 29
  runRoot: /run/user/1000/containers
  volumePath: /home/user/.local/share/containers/storage/volumes
version:
  APIVersion: 3.1.2
  Built: 1619097693
  BuiltTime: Thu Apr 22 20:21:33 2021
  GitCommit: ""
  GoVersion: go1.15.8
  OsArch: linux/amd64
  Version: 3.1.2
```

**Package info (e.g. output of `rpm -q podman` or `apt list podman`):**

```
podman-3.1.2-1.fc33.x86_64
```

**Playbok you run with ansible (e.g. content of `playbook.yaml`):**

```yaml
- name: Create network
  containers.podman.podman_network:
    name: test_network

- name: Create pod
  containers.podman.podman_pod:
    name: test_pod
    network: test_network
    ports:
      - '8000:80'

- name: Run test
  containers.podman.podman_container:
    name: test-container
    image: containous/whoami:latest
    pod: test_pod
    restart_policy: "unless-stopped"
    state: present
```

**Command line and output of ansible run with high verbosity**

**Please NOTE: if you submit a bug about idempotency, run the playbook with `--diff` option, like:**

`ansible-playbook -i inventory --diff -vv playbook.yml`

```
ansible-playbook 2.9.18
  config file = /home/user/production/ansible/ansible.cfg
  configured module search path = ['/home/user/.ansible/plugins/modules', '/usr/share/ansible/plugins/modules']
  ansible python module location = /usr/lib/python3.9/site-packages/ansible
  executable location = /usr/bin/ansible-playbook
  python version = 3.9.2 (default, Feb 20 2021, 00:00:00) [GCC 10.2.1 20201125 (Red Hat 10.2.1-9)]
Using /home/user/production/ansible/ansible.cfg as config file
Skipping callback 'actionable', as we already have a stdout callback.
Skipping callback 'counter_enabled', as we already have a stdout callback.
Skipping callback 'debug', as we already have a stdout callback.
Skipping callback 'dense', as we already have a stdout callback.
Skipping callback 'dense', as we already have a stdout callback.
Skipping callback 'full_skip', as we already have a stdout callback.
Skipping callback 'json', as we already have a stdout callback.
Skipping callback 'minimal', as we already have a stdout callback.
Skipping callback 'null', as we already have a stdout callback.
Skipping callback 'oneline', as we already have a stdout callback.
Skipping callback 'selective', as we already have a stdout callback.
Skipping callback 'skippy', as we already have a stdout callback.
Skipping callback 'stderr', as we already have a stdout callback.
Skipping callback 'unixy', as we already have a stdout callback.
Skipping callback 'yaml', as we already have a stdout callback.

PLAYBOOK: bug.yaml **************************************************************************************
1 plays in bug.yaml

PLAY [all] **********************************************************************************************

TASK [Gathering Facts] **********************************************************************************
task path: /home/user/production/ansible/bug.yaml:1
[DEPRECATION WARNING]: Distribution fedora 33 on host localhost should use /usr/bin/python3, but is 
using /usr/bin/python for backward compatibility with prior Ansible releases. A future Ansible release 
will default to using the discovered platform python for this host. See 
https://docs.ansible.com/ansible/2.9/reference_appendices/interpreter_discovery.html for more 
information. This feature will be removed in version 2.12. Deprecation warnings can be disabled by 
setting deprecation_warnings=False in ansible.cfg.
ok: [localhost]
META: ran handlers

TASK [bug : Create network] *****************************************************************************
task path: /home/user/production/ansible/roles/bug/tasks/main.yaml:1
ok: [localhost] => {"actions": [], "changed": false, "network": {"cniVersion": "0.4.0", "name": "test_network", "plugins": [{"bridge": "cni-podman1", "hairpinMode": true, "ipMasq": true, "ipam": {"ranges": [[{"gateway": "10.88.3.1", "subnet": "10.88.3.0/24"}]], "routes": [{"dst": "0.0.0.0/0"}], "type": "host-local"}, "isGateway": true, "type": "bridge"}, {"capabilities": {"portMappings": true}, "type": "portmap"}, {"backend": "", "type": "firewall"}, {"type": "tuning"}, {"capabilities": {"aliases": true}, "domainName": "dns.podman", "type": "dnsname"}]}, "podman_actions": [], "stderr": "", "stderr_lines": [], "stdout": "", "stdout_lines": []}

TASK [bug : Create pod] *********************************************************************************
task path: /home/user/production/ansible/roles/bug/tasks/main.yaml:6
fatal: [localhost]: FAILED! => {"changed": false, "msg": "Can't start pod test_pod", "stderr": "Error: error starting container 09ba64a54cd896529d4bcde82f1763b07638dea6460b6f26fe5cbecb402975de: a dependency of container 09ba64a54cd896529d4bcde82f1763b07638dea6460b6f26fe5cbecb402975de failed to start: container state improper\n", "stderr_lines": ["Error: error starting container 09ba64a54cd896529d4bcde82f1763b07638dea6460b6f26fe5cbecb402975de: a dependency of container 09ba64a54cd896529d4bcde82f1763b07638dea6460b6f26fe5cbecb402975de failed to start: container state improper"], "stdout": "error starting container c707755ec5ae467c4eaab80c3f691f7a1e462566bcacaf6d1ab381c543b2bdd5: command rootless-cni-infra [alloc c707755ec5ae467c4eaab80c3f691f7a1e462566bcacaf6d1ab381c543b2bdd5 test_network test_pod   ] in container 995ce4d5108faed62de42510df329d649674dc7af04fc9acc1d9c2a3dd18422f failed with status 1, stdout=\"\", stderr=\"failed to open netns \\\"/proc/10131/ns/net\\\": failed to Statfs \\\"/proc/10131/ns/net\\\": no such file or directory\\n\"\n", "stdout_lines": ["error starting container c707755ec5ae467c4eaab80c3f691f7a1e462566bcacaf6d1ab381c543b2bdd5: command rootless-cni-infra [alloc c707755ec5ae467c4eaab80c3f691f7a1e462566bcacaf6d1ab381c543b2bdd5 test_network test_pod   ] in container 995ce4d5108faed62de42510df329d649674dc7af04fc9acc1d9c2a3dd18422f failed with status 1, stdout=\"\", stderr=\"failed to open netns \\\"/proc/10131/ns/net\\\": failed to Statfs \\\"/proc/10131/ns/net\\\": no such file or directory\\n\""]}

PLAY RECAP **********************************************************************************************
localhost                  : ok=2    changed=0    unreachable=0    failed=1    skipped=0    rescued=0    ignored=0  
```

**Additional environment details (AWS, VirtualBox, physical, etc.):**

**Comment by sshnaidm** [comment]
I'll take a look, but seems to me podman problem, not collections or ansible.

**Comment by sshnaidm** [comment]
@phongngthanh 
I suggest to open issue in podman repo: https://github.com/containers/podman/issues because it's not issue in collection.
The reproducing is simple:

```
podman network create test_network
podman pod create --name test_pod --network test_network --publish 8000:80
podman run --name test-container --pod test_pod --restart=unless-stopped --detach=True containous/whoami:latest
```
reboot
```
podman rm -f test-container
podman run --name test-container --pod test_pod --restart=unless-stopped --detach=True containous/whoami:latest
```

The errors:
```
ERRO[0000] error starting some container dependencies   
ERRO[0000] "command rootless-cni-infra [alloc 5f9ef779c8c5418d11ff43427aebe7cc1e343bbb46a4d959f30a0302f7bdcca4 test_network test_pod   ] in container dfaa51e6e3a22d59563f496f1400f3763c8878a4ac0f8370288ae776b26a2fc8 failed with status 1, stdout=\"\", stderr=\"failed to open netns \\\"/proc/2261/ns/net\\\": failed to Statfs \\\"/proc/2261/ns/net\\\": no such file or directory\\n\"" 
Error: error starting some containers: internal libpod error
```

**Comment by phongngthanh** [comment]
I have opened the issues in the podman repo. Thanks for the assists @sshnaidm.
https://github.com/containers/podman/issues/10196

**Comment by sshnaidm** [comment]
@phongngthanh thanks, I see it's fixed in 3.2, so need just to wait for a new version.
